### DISCUSSION TITLE
pods restarts at k3s/kubelet restart/upgrade

### DISCUSSION THREAD

**Discussion by lictw** [k3s-io/k3s#7314] [Q&A] [answered]
Hello! 

While upgrading my clusters I saw that sometime all pods at node are restarting while sometime are not, is it normal behavior related to specific version that should recreate all containers to achieve something and sometimes not, or pods strictly should or should not be restarted while upgrade?

**Comment by brandond** [comment, marked as answer]
It really depends on what version you're upgrading from and to. In general, pods should remain running and not be restarted when K3s is stopped and started. If you are doing more than stopping and starting - performing an upgrade that requires changes to the pod sandbox - that may cause pods to be recreated.

**Reply by lictw** [reply]
Thanks! Can I found somewhere information about will current version restart pods or not?

**Reply by brandond** [reply]
There's not really a good way to know, sorry. If in doubt, plan for K3s upgrades to possibly trigger pod restarts.

Note: *UPGRADES*, not just restarts of the service.

**Reply by lictw** [reply]
Got it, hope last question: I saw pods restarts on all clusters when was upgrading 1.22 -> 1.23, and no restarts when 1.23 -> 1.24, but when I tested upgrade 1.22 -> 1.24, there was no restarts too, but I expected restarts, if something was changed in 1.23 that require restarts, that minimum the same should be in 1.24, why no restarts? Is it about the reason why I shouldn't skip releases and should upgrade from one big version to next?

**Reply by brandond** [reply]
It is not supported to skip minors. It's possible you're missing something that's not being picked up because of an unexpected jump in versions.
